core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2765,6 +2765,11 @@ object SparkContext extends Logging {
   private[spark] val RDD_SCOPE_KEY = "spark.rdd.scope"
   private[spark] val RDD_SCOPE_NO_OVERRIDE_KEY = "spark.rdd.scope.noOverride"
 
+  /**
+   * Statement id is only used for thrift server
+   */
+  private[spark] val SPARK_STATEMENT_ID = "spark.statement.id"
```

Comment:
We can define it in sql/core module, as it's only used there and the STS module. Probably in `SQLExecution` object.